src/codeCache.cpp

```diff
@@ -124,6 +138,19 @@ const char* NativeCodeCache::binarySearch(const void* address) {
     if (low > 0 && (_blobs[low - 1]._start == _blobs[low - 1]._end || _blobs[low - 1]._end == address)) {
         return (const char*)_blobs[low - 1]._method;
     }
+
+    if (_add_build_ids && _build_id) {
+        char buf[512];
+        // add 1 for the null terminator
+        const int n = 1 + snprintf(buf, sizeof(buf), "%s %s+%p_[bid]", _name, _build_id, (void*)((char*)address - (char*)_min_address));
+        char *allocated;
+        // if snprintf didn't overflow, and we manage to allocate a suitable buffer (from the LinearAllocator, not malloc,
+        // since this is signal context and malloc can deadlock) - then use them.
+        if (n < sizeof(buf) && (allocated = (char*)allocator->alloc(n)) != NULL) {
```

Comment:
I wanna do another round at verifying the lifetimes here (the pointer returned by `alloc()`)

Reply:
AFAICT, `LinearAllocator` is `clear`ed in two places:
* `LinearAllocator`'s destructor, which is called from `Profiler`'s destructor, which is a static variable so destructed only upon `async-profiler`'s unload .
* `CallTraceStorage::clear`, which is called from `Profiler::start`, which is called on a `start` command.

So I think you're good.

Reply:
Sounds right. Thanks for the double check.